src/browser/api/window.js

```diff
@@ -1502,18 +1502,60 @@ Window.moveTo = function(identity, left, top) {
 };
 
 Window.navigate = function(identity, url) {
-    let browserWindow = getElectronBrowserWindow(identity, 'navigate');
-    browserWindow.webContents.loadURL(url);
+    return new Promise((resolve, reject) => {
+        const browserWindow = getElectronBrowserWindow(identity);
+
+        if (!browserWindow) {
```

Comment:
You can delete these checks, we have middleware to check [entity existence](https://github.com/HadoukenIO/core/blob/develop/src/browser/api_protocol/api_handlers/middleware_entity_existence.ts)